### PR TITLE
Support ARM layers

### PR DESCRIPTION
**Pull request by duncanista**
### What does this PR do?

Add support for ARM layers. Check [SLS-1387](https://datadoghq.atlassian.net/browse/SLS-1387).
Swap previous layer if lambda layers already contain a previous version.

### Motivation

Follow up with 

### Testing Guidelines

- [x] Added tests to add correct layer given the architecture.
- [x] Ignore architecture if runtime doesn't require us to change the layer. 

### Additional Notes

Upgraded SLS Framework to latest version.

### Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: integration_tests/correct_extension_apigateway_snapshot.json [hunk 1]
```diff
@@ -858,7 +861,10 @@
           }
         },
         "MethodResponses": []
-      }
+      },
+      "DependsOn": [
```
[the remark this answers is not in the document]
Comment (duncanista) [reply integration_tests/correct_extension_apigateway_snapshot.json:865]: Non-related to dashcon, the dependency bump requires the DependsOn.

File: src/index.ts [hunk 2]
```diff
@@ -17,16 +17,9 @@ import { TracingMode, enableTracing } from "./tracing";
 import { redirectHandlers } from "./wrapper";
 import { addCloudWatchForwarderSubscriptions, addExecutionLogGroupsAndSubscriptions } from "./forwarder";
 import { addOutputLinks, printOutputs } from "./output";
-import { FunctionDefinition } from "serverless";
 import { setMonitors } from "./monitors";
 import { getCloudFormationStackId } from "./monitor-api-requests";
 
-// Separate interface since DefinitelyTyped currently doesn't include tags or env
```
Comment (jcstorms1) [inline src/index.ts:24]: Curious about why we can delete the interface here. Is it unnecessary?

**Review by jcstorms1 — COMMENTED** [review]
Hey @duncanista this looks good to me but I'd get a second pair of eyes on it if that's OK.

File: integration_tests/correct_extension_apigateway_snapshot.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (DarcyRaynerDD) [reply integration_tests/correct_extension_apigateway_snapshot.json:865]: I think there is a - in the function names, and it gets translated to 'Dash' by the framework

File: src/index.ts (hunk 2 above)
Comment (duncanista) [reply src/index.ts:24]: The version we currently have of DefinintelyTyped now supports it. I guess the interface was created when it wasn't.

**Review by maxday — APPROVED** [review]
LGTM !

**Pull request merged**